worker/sshserver/worker.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver
+
+import (
+	"context"
+
+	"github.com/gliderlabs/ssh"
+	"github.com/juju/errors"
+	"github.com/juju/worker/v3"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/state"
+)
+
+// sshServerWorker is a worker that runs an embedded SSH server.
+// See the server implementation for more details on how it works.
+type sshServerWorker struct {
+	tomb tomb.Tomb
+
+	srv *ssh.Server
+}
+
+// NewSSHServerWorker returns a new worker that runs an embedded SSH server.
+func NewSSHServerWorker(statePool *state.StatePool, jumpHostKey string) (worker.Worker, error) {
```

Comment:
This should be tested, even if you have a generic `worker_test.go`. See other workers.

Reply:
Agreed, I'll try add one, I noticed other workers put the bulk into worker.go but I wanted to separate it into server.go (felt cleaner in my head). I'll make a generic test for this though.

Reply:
It's now got a nested worker, but wasn't sure how to test worker.go now... I did see workertest package but yeah.. Not 100% on testing this.